File: 프로그래머스/2/77485. 행렬 테두리 회전하기/행렬 테두리 회전하기.java
class Solution {
    public int[] solution(int rows, int columns, int[][] queries) {
        int[][] matrix = new int[rows][columns];
        int[] answer = new int[queries.length];
        int count = 1;

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns; j++) {
                matrix[i][j] = count++;
            }
        }
        
        for (int i = 0; i < queries.length; i++) {
            answer[i] = rotate(matrix, queries[i][0], queries[i][1], queries[i][2], queries[i][3]);
        }
        
        return answer;
    }
    
    public static int rotate(int[][] matrix, int x1, int y1, int x2, int y2) {
        int temp = matrix[x1 - 1][y1 - 1];
        int minVal = temp;

        for (int i = y1; i < y2; i++) {
            int current = matrix[x1 - 1][i];
            matrix[x1 - 1][i] = temp;
            temp = current;
            minVal = Math.min(minVal, temp);
        }

        for (int i = x1; i < x2; i++) {
            int current = matrix[i][y2 - 1];
            matrix[i][y2 - 1] = temp;
            temp = current;
            minVal = Math.min(minVal, temp);
        }

        for (int i = y2 - 2; i >= y1 - 1; i--) {
            int current = matrix[x2 - 1][i];
            matrix[x2 - 1][i] = temp;
            temp = current;
            minVal = Math.min(minVal, temp);
        }

        for (int i = x2 - 2; i >= x1 - 1; i--) {
            int current = matrix[i][y1 - 1];
            matrix[i][y1 - 1] = temp;
            temp = current;
            minVal = Math.min(minVal, temp);
        }

        return minVal;
    }
    
}

File: 프로그래머스/2/77485. 행렬 테두리 회전하기/행렬 테두리 회전하기.py
def solution(rows, columns, queries):
    matrix = [[(i - 1) * columns + j for j in range(1, columns + 1)] for i in range(1, rows + 1)]
    result = []

    for query in queries:
        result.append(rotate(matrix, query[0], query[1], query[2], query[3]))

    return result

def rotate(matrix, x1, y1, x2, y2):
    temp = matrix[x1 - 1][y1 - 1]
    min_val = temp

    for i in range(y1, y2):
        matrix[x1 - 1][i], temp = temp, matrix[x1 - 1][i]
        min_val = min(min_val, temp)

    for i in range(x1, x2):
        matrix[i][y2 - 1], temp = temp, matrix[i][y2 - 1]
        min_val = min(min_val, temp)

    for i in range(y2 - 2, y1 - 2, -1):
        matrix[x2 - 1][i], temp = temp, matrix[x2 - 1][i]
        min_val = min(min_val, temp)

    for i in range(x2 - 2, x1 - 2, -1):
        matrix[i][y1 - 1], temp = temp, matrix[i][y1 - 1]
        min_val = min(min_val, temp)

    return min_val